src/main/java/teammates/logic/core/GoogleCloudLoggingService.java

```diff
@@ -108,6 +120,45 @@
         return errorLogs;
     }
 
+    @Override
+    public QueryLogsResults queryLogs(List<String> severities, Instant startTime, Instant endTime,
+            Integer pageSize, String pageToken) throws LogServiceException {
+        LogSearchParams logSearchParams = new LogSearchParams()
+                .addLogName(STDOUT_LOG_NAME)
+                .addLogName(STDERR_LOG_NAME)
+                .setSeverities(severities)
+                .setStartTime(startTime)
+                .setEndTime(endTime);
+
+        PageParams pageParams = new PageParams(pageSize, pageToken);
+
+        Page<LogEntry> logEntriesInPage = getLogEntries(logSearchParams, pageParams);
+        List<GeneralLogEntry> logEntries = new ArrayList<>();
+        for (LogEntry entry : logEntriesInPage.getValues()) {
+            String logName = entry.getLogName();
+            Severity severity = entry.getSeverity();
+            String trace = entry.getTrace();
+            com.google.cloud.logging.SourceLocation sourceLocation = entry.getSourceLocation();
+            Payload<?> payload = entry.getPayload();
+            long timestamp = entry.getTimestamp();
+
+            GeneralLogEntry logEntry;
+            if (payload.getType() == Payload.Type.JSON) {
+                JSONObject jsonObject = new JSONObject(((Payload.JsonPayload) payload).getDataAsMap());
```

Comment:
Unless there's specific reasons/methods of `JSONObject` class that is being used here, I don't see why the need for wrapping it when leaving it as `Map<String, Object>` works.

Reply:
Ok. Will change this to map